lms/djangoapps/discussion_api/api.py

```diff
@@ -415,7 +415,7 @@ def get_comment_list(request, thread_id, endorsed, page, page_size, mark_as_read
         raise Http404
     num_pages = (resp_total + page_size - 1) / page_size if resp_total else 1
 
-    results = [CommentSerializer(response, context=context).data for response in responses]
+    results = [CommentSerializer(response, remove_fields=["children"], context=context).data for response in responses]
```

Comment:
@nasthagiri In this case; we need to explicitly remove children field as the serializable object (i.e. `response` in this line of code) contains non null children list.
However, other usage of `remove_field` that we initially introduced for `response_count` could be refactored using `required=False` (that is now available in updated DRF version). I am going to create a new PR for that and link JIRA task that we created for it.